resolver/rewriter_resolver_test.go

```diff
@@ -126,6 +141,54 @@ var _ = Describe("RewriterResolver", func() {
 				return mNextResponse, nil
 			}
 		})
+
+		It("should not call next resolver", func() {
+			fqdnOriginal = sampleOriginal
+			fqdnRewritten = sampleRewritten
+			expectNilAnswer = true
+
+			// Make inner return a nil Answer but not an empty Response
+			mInner.ResolveFn = func(req *model.Request) (*model.Response, error) {
+				Expect(req).Should(Equal(request))
+
+				// Inner should see fqdnRewritten
+				Expect(req.Req.Question[0].Name).Should(Equal(fqdnRewritten))
+
+				return &model.Response{Res: &dns.Msg{Question: req.Req.Question, Answer: nil}}, nil
+			}
+
+			// Resolver after RewriterResolver should not be called `fqdnOriginal`
+			mNext.AssertNotCalled(GinkgoT(), "Resolve", mock.Anything)
+		})
+
+		When("has fallback on failure", func() {
```

Comment:
The old name is left here: "fallback on failure".

Reply:
Done!